File: solutions/Insert-Interval/Insert-Interval.cpp

/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */
class Solution {
/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */
class Solution {
public:
    vector<Interval> insert(vector<Interval> &intervals, Interval newInterval) {
        vector<Interval> ans;
        int size=intervals.size();
        for(int i=0;i<size;i++)
            if(newInterval.start>intervals[i].end)
                ans.push_back(intervals[i]);
            else if(newInterval.end<intervals[i].start)
            {
                ans.push_back(newInterval);
                ans.insert(ans.end(),intervals.begin()+i,intervals.end());
                return ans;
            }
            else
            {
                newInterval.start=min(newInterval.start,intervals[i].start);
                newInterval.end=max(newInterval.end,intervals[i].end);
            }
        ans.push_back(newInterval);
        return ans;
    }
};


 

File: solutions/Insert-Interval/Insert-Interval.js
/**
 * Definition for an interval.
 * function Interval(start, end) {
 *     this.start = start;
 *     this.end = end;
 * }
 */
/**
 * @param {Interval[]} intervals
 * @param {Interval} newInterval
 * @return {Interval[]}
 */
var insert = function (intervals, newInterval) {
  const n = intervals.length;
  let ans = [];
  for (let i = 0; i < n; i++) {
    if (newInterval.start > intervals[i].end) {
      ans.push(intervals[i]);
    } else if (newInterval.end < intervals[i].start) {
      ans.push(newInterval);
      ans = ans.concat(intervals.slice(i));
      return ans;
    } else {
      newInterval.start = Math.min(newInterval.start, intervals[i].start);
      newInterval.end = Math.max(newInterval.end, intervals[i].end);
    }
  }
  ans.push(newInterval);
  return ans;
};